go/libraries/doltcore/sqle/sqlddl_test.go

```diff
@@ -1186,6 +1186,90 @@ INSERT INTO child_non_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5'
 	}
 }
 
+func TestDropIndex(t *testing.T) {
```

Comment:
Probably better expressed as a Sql script test in ddl_queries.go

Reply:
I don't think there are any queries that show what underlying secondary index a foreign key is using, which is why I wrote the tests here.

Reply:
I missed the explicit checks into foreign keys here, my fault.

There are some decent functional tests in GMS for this, so I think you're covered there.

But you don't have coverage for a bunch of scenarios you should be testing. Specifically, there aren't multiple indexes to choose from when you drop the existing one here.

A unit test for findIndexWithPrefix could be more effective for that, your call